### PR TITLE
Base icon component

### PR DESCRIPTION
**Pull request by WrathOfZombies**
- [x] Port base component
- [x] Add custom prop types & docs
- [x] Add examples
- [x] Add variants
- [x] Add states
- [ ] Add knobs - skipping this do this due to lack of component type

### CODE REVIEW

File: src/components/Icon/Icon.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    //   useKeyOnly(disabled, 'disabled'),
+    //   useKeyOnly(fitted, 'fitted'),
+    //   useKeyOnly(inverted, 'inverted'),
+    //   useKeyOnly(link, 'link'),
+    //   useKeyOnly(loading, 'loading'),
+    //   useValueAndKey(flipped, 'flipped'),
+    //   useValueAndKey(rotated, 'rotated'),
+    //   'icon',
+    //   className,
+    // )
```
Comment (levithomason) [inline src/components/Icon/Icon.js]: Safe to delete all the cruft here.  We'll reimplement what we need as we need it.

File: src/components/Icon/Icon.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    // name: customPropTypes.suggest(SUI.ALL_ICONS_IN_ALL_CONTEXTS),
+
+    /** Icon can rotated. */
+    rotated: PropTypes.oneOf(['clockwise', 'counterclockwise']),
+
+    /** Size of the icon. */
+    // size: PropTypes.oneOf(_.without(SUI.SIZES, 'medium')),
+
+    /** Icon can have an aria label. */
+    'aria-label': PropTypes.string,
```
Comment (levithomason) [inline src/components/Icon/Icon.js]: Let's strip propTypes down to only what this PR will add.  Preferably, the bare minimum needed to render the default example.  Probably the `name` prop?

We'll add props and examples as we need them.

**Review by levithomason — CHANGES_REQUESTED** [review]
Let's split a PR out from here that implements only the default icon example.  There's a bit too much to review effectively in this PR.

Ideally, the first PR will only be the base implementation.  Then, each prop/feature will be added as needed.

**Comment by kuzhelov** [comment]
taking this PR to split it on two different ones

**Comment by levithomason** [comment]
See the new repo Icon.  We've trimmed it down and updated it there for our initial needs.

https://github.com/stardust-ui/react

**Pull request closed without merging**